tests/bind.rs

```diff
@@ -0,0 +1,92 @@
+mod fixtures;
+
+use assert_cmd::prelude::*;
+use assert_fs::fixture::TempDir;
+use fixtures::{port, server, tmpdir, Error, TestServer};
+use regex::Regex;
+use rstest::rstest;
+use std::io::{BufRead, BufReader};
+use std::process::{Command, Stdio};
+use std::thread::sleep;
+use std::time::Duration;
+
+#[rstest]
+#[case(&["-i", "012.123.234.012"])]
+#[case(&["-i", "::", "-i", "012.123.234.012"])]
+fn bind_fails(tmpdir: TempDir, port: u16, #[case] args: &[&str]) -> Result<(), Error> {
+    Command::cargo_bin("miniserve")?
+        .arg(tmpdir.path())
+        .arg("-p")
+        .arg(port.to_string())
+        .args(args)
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
```

Comment:
Check for the actual error returned here. Not all of it should be a OS error. For instance, it should be possible to check `stderr` for at least `"Failed to bind server"`.